beacon_node/lighthouse_network/src/rpc/self_limiter.rs

```diff
@@ -13,30 +20,31 @@ use tokio_util::time::DelayQueue;
 use tracing::debug;
 use types::{EthSpec, ForkContext};
 
-use super::{
-    config::OutboundRateLimiterConfig,
-    rate_limiter::{RPCRateLimiter as RateLimiter, RateLimitedErr},
-    BehaviourAction, Protocol, RPCSend, ReqId, RequestType,
-};
-
 /// A request that was rate limited or waiting on rate limited requests for the same peer and
 /// protocol.
 struct QueuedRequest<Id: ReqId, E: EthSpec> {
     req: RequestType<E>,
     request_id: Id,
+    queued_at: Duration,
 }
 
+/// The number of milliseconds requests delayed due to the concurrent request limit stay in the queue.
+const WAIT_TIME_DUE_TO_CONCURRENT_REQUESTS: u64 = 100;
+
+#[allow(clippy::type_complexity)]
 pub(crate) struct SelfRateLimiter<Id: ReqId, E: EthSpec> {
-    /// Requests queued for sending per peer. This requests are stored when the self rate
+    /// Active requests that are awaiting a response.
+    active_requests: HashMap<PeerId, HashMap<Protocol, usize>>,
+    /// Requests queued for sending per peer. These requests are stored when the self rate
     /// limiter rejects them. Rate limiting is based on a Peer and Protocol basis, therefore
     /// are stored in the same way.
     delayed_requests: HashMap<(PeerId, Protocol), VecDeque<QueuedRequest<Id, E>>>,
```

Comment:
@ackintosh can you add new metrics here for:
- Counter: Total count of queue requests labeled by `Protocol` string
- Histogram:  `VecDeque` length at the moment of queuing a request, labeled by `Protocol` string = observe length before pushing
- Gauge: Current sum of `VecDeque` lengths for all items in `delayed_requests`